### PR TITLE
MC-26946: added fullScreenLoader.stopLoader method to validation

### LINKED ISSUE #26946: Declined Braintree 3DS enabled transactions result in stuck loader icon
### Preconditions (*)
1. 2.4-develop

### Steps to reproduce (*)
```
Console
=======
deploy 2.4-develop from git

Backend
=======
Catalog -> Products
    Add Product
        Name = Test
        Price = 2000 <- causes declined transaction
        Quantity = 100
        New Category
            Category Name = tc
            Parent Category = Default Category
            Create Category
        Save & Close
Store -> Configuration
	Sales -> Payment Methods
		Merchant Country = United Kingdom
		Braintree -> Configure
			Payment Action = Authorize and Capture
			Merchant ID = xxxx
			Public Key = xxxx
			Private Key = xxxx
			Enable this solution = Yes
			3DSecure Verification = Yes
			Debug = Yes
	Save Config
Clear Cache

Frontend
========
View Test Product
Add to Cart
Proceed to checkout
	Enter email address
	Enter address details
	Choose Fixed flat rate shipping
Click Next
	Credit Card (Braintree)
		4111111111111111
		01/21
		123
	Place Order
	Enter 1234 + Submit on 3DS screen
```
### Expected result (*)
1. Transaction is declined and able to retry using different details

### Actual result (*)
1. Checkout is stuck on loader

### PR DESCRIPTION
**Pull request by AleksLi**
### Description (*)

3d-secure.js **validate** method in one of the scenarios doesn't stopLoader and because of that it was impossible to make any further changes from the user perspective.

### Fixed Issues (if relevant)

1. magento/magento2#26946: Declined Braintree 3DS enabled transactions result in stuck loader icon

### Manual testing scenarios (*)

1. well Described inside the ticket #26946

### Questions or comments

I'm not sure if we need any tests for that issue.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/Braintree/view/frontend/web/js/view/payment/3d-secure.js [hunk 1]
```diff
@@ -157,9 +157,9 @@ define([
                             context.paymentPayload.nonce = payload.nonce;
                             self.state.resolve();
                         } else {
-                            fullScreenLoader.stopLoader();
                             self.state.reject($t('Please try again with another form of payment.'));
                         }
+                        fullScreenLoader.stopLoader();
                     });
                 })
                 .fail(function () {
```
Comment (pmclain) [inline app/code/Magento/Braintree/view/frontend/web/js/view/payment/3d-secure.js:163]: What are your thoughts on adding an `.always(function () {  fullScreenLoader.stopLoader(); });` after this `.fail` to replace the `fullScreenLoader.stopLoader()` in every conditional path?

**Review by pmclain — DISMISSED** [review]

File: app/code/Magento/Braintree/view/frontend/web/js/view/payment/3d-secure.js (hunk 1 above)
Comment (AleksLi) [reply app/code/Magento/Braintree/view/frontend/web/js/view/payment/3d-secure.js:163]: Yes, I think that's a good way to refactor this part of code.
Thank you for the suggestion. I will update my PR now.

**Review by pmclain — DISMISSED** [review]

**Comment by magento-engcom-team** [comment]
Hi @pmclain, thank you for the review.
[ENGCOM-6989](https://jira.corp.magento.com/browse/ENGCOM-6989) has been created to process this Pull Request
:eight_spoked_asterisk: @pmclain, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Review by engcom-Alfa — CHANGES_REQUESTED** [review]
Hi @AleksLi  .
     Due to [Magento Definition of Done](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing_dod.html) all code must be covered by tests. Please cover your fix by automated test.
Thanks!

**Comment by AleksLi** [comment]
@engcom-Alfa what should I do. What should I cover? Could you explain this for me a bit. The tests are green.

**Comment by engcom-Alfa** [comment]
Hi @AleksLi.
 Due to [Magento Definition of Done](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing_dod.html) all code must be covered by tests. You can cover this PR by jasmine test. ([JavaScript unit testing with Jasmine](https://devdocs.magento.com/guides/v2.3/test/js/jasmine.html)).
Thanks!

**Comment by AleksLi** [comment]
@engcom-Alfa should I cover the whole file? Because I cannot find 3d-secure.test.js in Braintree directory. I searched in `dev/tests/js/jasmine/tests/app/code/Magento` folder.

**Comment by AleksLi** [comment]
@engcom-Alfa Could someone help me to cover this file with jasmine tests?

**Comment by engcom-Alfa** [comment]
> @engcom-Alfa Could someone help me to cover this file with jasmine tests?

@engcom-Golf will help you.

**Comment by AleksLi** [comment]
@engcom-Alfa  Please write to me in the slack if you have any questions. I'm open for conversation and your help.

My name in Magento slack channel is **@aleksliwork (Aleks Litvinov).**

How we can do this? 
I'm open for your help.

**Comment by Nazar65** [comment]
HI @AleksLi @engcom-Echo will help you with test coverage.

**Comment by engcom-Echo** [comment]
I will help with the coverage of the tests

**Comment by engcom-Echo** [comment]
Hi @AleksLi, I'm closing this PR. The code for this module has been removed from the branch.
Please reopen and update if you wish to continue.
Thank you for your collaboration.

**Pull request closed without merging**